globals/card_definitions.gd

```diff
@@ -4,33 +4,25 @@ var card_data = []
 
 var card_definitions_path = "res://data/card_definitions.json"
 var decks_path = "res://data/decks"
-var decks = []  # An array of (JSON) dictionaries
+var decks = {}  # A dictionary of (JSON) dictionaries
 
 const CardHighlightColor = "#7DF9FF" # Light blue
 
 func get_deck_test_deck():
-	for deck in decks:
-		if deck['id'] == "rachel":
-			return deck
-	return get_random_deck(-1)
+	return decks.get("rachel", get_random_deck(-1))
 
 func get_random_deck(season : int):
 	# Randomize
 	if season == -1:
-		var random_index = randi() % len(decks)
-		while decks[random_index]['id'] in GlobalSettings.CharacterBanlist:
-			random_index = randi() % len(decks)
-		return decks[random_index]
+		var deck_names = decks.keys()
+		var random_deck = deck_names.pick_random()
+		while random_deck in GlobalSettings.CharacterBanlist:
+			random_deck = deck_names.pick_random()
+		return decks[random_deck]
 	else:
-		var season_decks = []
-		for deck in decks:
-			if deck['id'] in GlobalSettings.CharacterBanlist:
-				continue
-			if deck['season'] == season:
-				season_decks.append(deck)
-		var random_index = randi() % len(season_decks)
-		return season_decks[random_index]
-
+		var season_decks = decks.values().filter(func (deck):
```

Comment:
It probably makes more sense to do this above to filter out the banlist in a similar way for the -1 season.

Reply:
Good call. Fixed and tested.